File: frontend/src/entity/BredPokemon.ts
import { Gender } from "./Gender";
import { Type } from "./Type";
import { Ability } from "./Ability";
import { Nature } from "./Nature";
import { Item } from "./Item";
import { BaseStats } from "./BaseStats";
import { Move } from './Move'; 
import { IndividualValues } from "./IndividualValues";
import { BasePoints } from "./BasePoints";
import { ActualValues } from "./ActualValues";

export interface BredPokemon {
    id: string;
    pokemonId: number;
    nationalPokedexNo: number;
    formeNo: number;
    name: string;
    formeName: string | null;
    gender: Gender;
    level: number;
    types: Type[];
    ability: Ability;
    nature: Nature;
    heldItem: Item | null;
    baseStats: BaseStats;
    individualValues: IndividualValues;
    basePoints: BasePoints;
    actualValues: ActualValues;
    moves: Move[];
    note: string | null;
}

File: backend/src/api_schema/bred_pokemon.go
package api_schema

type BredPokemon struct {
	Id                string           `json:"id"`
	PokemonId         int              `json:"pokemon_id"`
	NationalPokedexNo int              `json:"national_pokedex_no"`
	FormeNo           int              `json:"forme_no"`
	Name              string           `json:"name"`
	FormeName         *string          `json:"forme_name"`
	Gender            Gender           `json:"gender"`
	Level             int              `json:"level"`
	Types             []Type           `json:"types"`
	Ability           Ability          `json:"ability"`
	Nature            Nature           `json:"nature"`
	HeldItem          *Item            `json:"held_item"`
	BaseStats         BaseStats        `json:"base_stats"`
	IndividualValues  IndividualValues `json:"individual_values"`
	BasePoints        BasePoints       `json:"base_points"`
	ActualValues      ActualValues     `json:"actual_values"`
	Moves             []Move           `json:"moves"`
	Note              *string          `json:"note"`
}
